cmd/kube-apiserver/app/options/options.go

```diff
@@ -180,6 +181,10 @@ func (s *ServerRunOptions) Flags() (fss cliflag.NamedFlagSets) {
 		"A CIDR notation IP range from which to assign service cluster IPs. This must not "+
 		"overlap with any IP ranges assigned to nodes for pods.")
 
+	fs.IPNetVar(&s.SecondaryServiceClusterIPRange, "secondary-service-cluster-ip-range", s.SecondaryServiceClusterIPRange, ""+
```

Comment:
What's the reason for not having `--service-cluster-ip-range` accept a comma-delimited list and have the second (if exists) be the secondary range?

Reply:
The rational is the following [I have a feeling that it will be contentious point in the review :-) ]:
- There are so many reasons why a user would need multiple IPs(`multi networks`, `dual stack`, `multi legs NVA`, `multi physical nics on host acting as FE/BE style connections` and more.  
- There are none for multiple `3+` Service IPs ranges. User only needs two (and only for dual stack. As in, different families). Even for a multiple `PodIPs` (even same family `e.g. 2+ v4 PodIPs`). What is needed is not to support `n` ranges support. What is needed is an ability to have Service IP=> select the intended `PodIP` `beyond `IPFamily` selection`.  (Example Below)
- In context, services of type `ClusterIP, NodePort, and LoadBalancer` play a role in: 
   1. Ingress external traffic into the cluster: This is where we worry about which family of `ServiceIP` we are listening on *and* which `PodIP` do we route into. 
    2. Move traffic within the cluster (either for discoverability and/or for load balancing). And in both cases the what matter is the family of the `ServiceIP` and the selection of correct `PodIP`(as the below example). Not the IP of the service itself. 

## An example

> we will have to deal with this situation when `PodIPs` allow `multi v4, multi6 or multi mix > 2`.

Pod `p` has multiple v6 IPs. `PodIPs:2000:1, 3000:1, and 4000:1`. Only one of them `PodIP: [2000:1]` represents the `front end` of the pod, The other help the Pod to connect to various networks. As in egress only. Hence a service `s` will only need select that particular `PodIP:2000:1` because this is the pod ingress endpoint.  To take the example further. Let us assume that `p` wants to offer another ingress endpoint on a different `PodIP:[3000:1]` (e.g. first one was offered for external traffic, second is for local applications only). User can create a new service selecting the appropriate `PodIP` and `Service.Spec.Type == ClusterIP`, along with network policies. What will matter is allowing user to select the right `PodIP`, not the `Service IP` 

I am trying to solve for `current` visible requirements, not something we think will happen. I am open to a discuss and find use case that will ever need it.

Reply:
> I am trying to solve for current visible requirements, not something we think will happen. I am open to a discuss and find use case that will ever need it.

I agree with this approach, but having `--service-cluster-ip-range` to only accept two addresses (comma delimted) gives us the options to allow `n` addresses in the future _if_ the use-cases come up. We would have to deprecate/remove `secondary-service-cluster-ip-range` if we find that having more than 2 cluster IPs is useful.

Reply:
I agree with @andrewsykim here; this is also consistent with our handling of ClusterIPs where it's a comma-delimited list but only allows two values.

Reply:
I would +1 either changing the names (IE make each specific to an IP family) or going with the list approach.

Reply:
list approach has been applied, thanks.